step1/src/src/services.js

```diff
@@ -0,0 +1,118 @@
+import {state} from "./state.js.js";
+import {selectAll, selectOne, checklength } from "./utils.js.js"
+import {template } from "./template.js.js";
+
+
+const $app = selectOne("#app");
+const $appenderForm = selectOne('form[name="appenderForm"]', $app);
+const $modifiers = selectAll('.modifier', $app);
+const $modifierForm = selectOne('form[name="modifierForm"]', $app);
+const $deleters = selectAll(".deleter", $app);
+const $complete = selectAll(".complete", $app);
+const $cancellers = selectAll(".cancel", $app);
+
+//이벤트 등록
+
+
+
+const itemAdd = function (event) {
+
+    event
+        .preventDefault();
+    const content = selectOne('input', $appenderForm)
+        .value
+        .trim();
+    checklength(content);
+
+    state.todoItems.push({
+        createdAt: Date.now(),
+        content: content,
+        isComplete: false
+    });
+    render();
+}
+const editItem = function (event) {
+    state
+        .selectedItem = Number(event.target.dataset.key);
+    render();
+  }
+  //아이템 수정
+const updateItem = function (event) {
+    event.preventDefault();
+    const content = selectOne('input', $modifierForm)
+        .value
+        .trim();
+    checklength(content);
+    state
+        .todoItems[state.selectedItem]
+        .content = content;
+    state
+        .selectedItem = -1;
+    render();
+  }
+  //아이템 수정 취소, esc키 //
+const resetItem = function(event){
+    event
+        .preventDefault();
+    state
+        .selectedItem = -1;
+    
+    render();
+  }
+  //아이템 삭제 관리//
+const deleteItem = function (event) {
+    const key = Number(event.target.dataset.key);
+    state
+        .todoItems
+        .splice(key, 1);
+    render();
+  }    
+  
+const toggleItem = function (event) {
+    const key = Number(event.target.dataset.key);
+    const item = state
+        .todoItems[key];
+    item
+        .isComplete = !item
+            .isComplete;
+    render();
+}
+export function render() {
+   
+    $app.innerHTML = template();
+
+    
+    $appenderForm
+        .addEventListener('submit', itemAdd); // 이쪽에서 addEventListner null 에러가 납니다
+    
+    
+    $modifiers
+        .forEach(($modifier) => $modifier
+            .addEventListener("click", editItem));
+    
+  
+    if ($modifierForm) {
+        $modifierForm
+            .addEventListener('submit', updateItem);
+        $modifierForm.
+            addEventListener('reset', resetItem);
```

Comment:
```suggestion
        $modifierForm.addEventListener('submit', updateItem);
        $modifierForm.addEventListener('reset', resetItem);
```
`한 줄에 점 하나` 이기 때문에 이렇게 표현하는게 더 자연스럽답니다!

Reply:
다른 코드도 이런 형태로 수정해주세요~